From ba829a18ace81c1e1f06b1a93eb31b294f0ba729 Mon Sep 17 00:00:00 2001
From: Sergio Lopez <slp@redhat.com>
Date: Mon, 20 Jan 2025 16:38:26 +0100
Subject: [PATCH 22/22] arch/x86,efi/libstub: use -std=gnu11 flag

GCC 15 defaults to gnu23, but the kernel is not compatible to that
standard. Other parts of the kernel use -std=gnu11, so let's do the same
in arch/x86/boot/compressed and efi/libstub.

Signed-off-by: Sergio Lopez <slp@redhat.com>
---
 arch/x86/boot/compressed/Makefile     | 2 +-
 drivers/firmware/efi/libstub/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/boot/compressed/Makefile b/arch/x86/boot/compressed/Makefile
index f2051644de94..5c8b86dfb1ea 100644
--- a/arch/x86/boot/compressed/Makefile
+++ b/arch/x86/boot/compressed/Makefile
@@ -24,7 +24,7 @@ targets := vmlinux vmlinux.bin vmlinux.bin.gz vmlinux.bin.bz2 vmlinux.bin.lzma \
 # case of cross compiling, as it has the '--target=' flag, which is needed to
 # avoid errors with '-march=i386', and future flags may depend on the target to
 # be valid.
-KBUILD_CFLAGS := -m$(BITS) -O2 $(CLANG_FLAGS)
+KBUILD_CFLAGS := -std=gnu11 -m$(BITS) -O2 $(CLANG_FLAGS)
 KBUILD_CFLAGS += -fno-strict-aliasing -fPIE
 KBUILD_CFLAGS += -Wundef
 KBUILD_CFLAGS += -DDISABLE_BRANCH_PROFILING
diff --git a/drivers/firmware/efi/libstub/Makefile b/drivers/firmware/efi/libstub/Makefile
index ed4e8ddbe76a..78cdf380ffcc 100644
--- a/drivers/firmware/efi/libstub/Makefile
+++ b/drivers/firmware/efi/libstub/Makefile
@@ -36,7 +36,7 @@ cflags-$(CONFIG_LOONGARCH)	+= -fpie
 cflags-$(CONFIG_EFI_PARAMS_FROM_FDT)	+= -I$(srctree)/scripts/dtc/libfdt
 
 KBUILD_CFLAGS			:= $(subst $(CC_FLAGS_FTRACE),,$(cflags-y)) \
-				   -Os -DDISABLE_BRANCH_PROFILING \
+				   -std=gnu11 -Os -DDISABLE_BRANCH_PROFILING \
 				   -include $(srctree)/include/linux/hidden.h \
 				   -D__NO_FORTIFY \
 				   -ffreestanding \
-- 
2.45.2

